lib/src/main/java/tech/relaycorp/relaydroid/messaging/SendMessage.kt

```diff
@@ -0,0 +1,93 @@
+package tech.relaycorp.relaydroid.messaging
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import tech.relaycorp.poweb.PoWebClient
+import tech.relaycorp.relaydroid.PrivateThirdPartyEndpoint
+import tech.relaycorp.relaydroid.PublicThirdPartyEndpoint
+import tech.relaycorp.relaydroid.Relaynet
+import tech.relaycorp.relaydroid.RelaynetException
+import tech.relaycorp.relaydroid.common.Logging.logger
+import tech.relaycorp.relaynet.bindings.pdc.Signer
+import tech.relaycorp.relaynet.issueEndpointCertificate
+import tech.relaycorp.relaynet.messages.Parcel
+import tech.relaycorp.relaynet.wrappers.x509.Certificate
+import java.util.logging.Level
+import kotlin.coroutines.CoroutineContext
+
+internal class SendMessage(
+    private val poWebClientBuilder: () -> PoWebClient =
+        { PoWebClient.initLocal(Relaynet.POWEB_PORT) },
+    private val coroutineContext: CoroutineContext = Dispatchers.IO
+) {
+
+    suspend fun send(message: OutgoingMessage) {
+        withContext(coroutineContext) {
+            message.validate()
```

Comment:
I think we're not taking full advantage of the fact we're wrapping the outgoing message in its object if we're deferring its validation until we're about to send it.

I think a better time to validate it would be when the Android dev creates an `OutgoingMessage`.

Reply:
This hasn't been changed in the last commit: `OutgoingMessage.build()` is still called within `SendMessage.send()`.

What I'm suggesting is essentially to call `Parcel(...)` in the constructor for `OutgoingMessage`, and store the resulting parcel. That way, `SendMessage.send()` would simply have to do something like `message.build(senderPrivateKey)` -- which would then internally call `parcel.serialize(privateKey)`. 

Also note that `Parcel.serialize()` won't raise any exceptions. Unless, of course, there's a bug in the code. But in principle, we shouldn't try to catch any exceptions when calling that method.